src/main/java/org/broadinstitute/hellbender/engine/spark/SparkContextFactory.java

```diff
@@ -103,21 +117,24 @@ public static synchronized void stopSparkContext(final JavaSparkContext context)
      * setup a spark context with the given name, master, and settings
      */
     @VisibleForTesting
-    static SparkConf setupSparkConf(final String appName, final String master, final Map<String, String> sparkAttributes) {
+    static SparkConf setupSparkConf(final String appName, final String master, final Map<String, String> suggestedAttributes, final Map<String,String> overridingAttributes) {
```

Comment:
So we have suggested attributes, mandatory attributes, overriding attributes, CMDLINE_ATTRIBUTES, and TEST_ATTRIBUTES? My head hurts! :)